src/inference/graph_inference.cpp

```diff
@@ -432,9 +426,9 @@ class GraphNodeAssignment : public Nice2Assignment {
     }
 
     for (size_t i = 0; i < query_->factors_of_a_node_[node].size(); ++i) {
-      uint64 hash = HashInt(assignments_[node].label);
-      for (auto var_it = query_->factors_of_a_node_[node][i].begin();
-                var_it != query_->factors_of_a_node_[node][i].end(); ++var_it) {
+      uint64 hash = 0;
+      for (auto var_it = query_->factors_of_a_node_[node][i]->begin();
```

Comment:
This is very unobvious code. What you want is to get a const vector<int>& factors_of_a_node = factors_of_a_node_[node]; and then iterate over the vector.